### DISCUSSION TITLE
Node names

### DISCUSSION THREAD

**Discussion by abrahami** [pyg-team/pytorch_geometric#2229] [Q&A]
Hi,
I have multiple graphs, taking from Reddit social network. Each node in each graph represents a Reddit user. I want to implement a graph classification solution. Indeed some of the nodes (users) are shared between different graphs.

When I am creating each graph in the package - should I keep the node index consistent between the different graphs? Is there any meaning to the node index in the graph?
As an example - when I create graphA with three nodes (1,2,3) and graphB with three nodes (3,4,5). Will the different algorithms take into account the fact that node 3 is shared between graphA and graphB or the index of the node has no meaning and only the X features are important?

thanks in advance!

**Comment by rusty1s** [comment]
Only the node features are important, so you may need to find a way to encode that information into the feature vector. The node indices are expected to go from 0 to N - 1.

**Reply by abrahami** [reply]
Great - thanks a lot!
Another question (somehow related).
Currently I don't have any node features (I only have information regarding the structure of the network). Is it possible to run classifiers this way?
I do have minimal information regarding edges (it is actually weight)

thanks again!

**Reply by rusty1s** [reply]
There are several options to generate node features in case you have none:
1. Use structural information such as node degree (via `torch_geometric.transforms.OneHotDegree`), useful for inductive tasks
2. Use node identities as features, either via the identity matrix, trainable embeddings or random embeddings, useful for transductive tasks.

**Reply by abrahami** [reply]
Got it, thank you so much!
I will try both options
